library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -31,12 +32,14 @@
                   List<DownloadFile> downloadFiles,
                   Map<DownloadFileId, Long> fileBytesDownloadedMap,
                   DownloadsBatchPersistence downloadsBatchPersistence,
-                  CallbackThrottle callbackThrottle) {
+                  CallbackThrottle callbackThrottle,
+                  DownloadConnectionAllowedChecker downloadConnectionAllowedChecker) {
```

Comment:
What about just `DownloadConnectionChecker` or even `ConnectionChecker`. `allowed` word is already available in `isAllowedToDownload` method which I think makes it clear.